packages/less/test/mocha-pptr/runner.js

```diff
@@ -0,0 +1,197 @@
+            }
+
+            !reporterIsChanged && m.setup({ 
+                reporter: Mocha.reporters[reporter] || Mocha.reporters.spec 
+            });
+
+            const runner = run(() => setTimeout(() => setResult.call(runner), 0))
+                .on('pass', test => { passes.push(test); all.push(test); })
+                .on('fail', test => { failures.push(test); all.push(test); })
+                .on('pending', test => { pending.push(test); all.push(test); })
+                .on('end', setResult);
+
+            return runner;
+        };
+    }
+
+    function shimMochaProcess(M) {
+        // Mocha needs a process.stdout.write in order to change the cursor position.
+        !M.process && (M.process = {});
+        !M.process.stdout && (M.process.stdout = {});
+
+        M.process.stdout.write = data => console.log('stdout:', data);
+        M.reporters.Base.useColors = true;
+        M.reporters.none = function None(runner) {
+            M.reporters.Base.call(this, runner);
+        };
+    }
+
+    Object.defineProperty(window, 'mocha', {
+        get: function() { return undefined },
+        set: function(m) {
+            shimMochaInstance(m);
+            delete window.mocha;
+            window.mocha = m
+        },
+        configurable: true
+    })
+
+    Object.defineProperty(window, 'Mocha', {
+        get: function() { return undefined },
+        set: function(m) {
+            shimMochaProcess(m);
+            delete window.Mocha;
+            window.Mocha = m;
+        },
+        configurable: true
+    });
+}
+
+function configureViewport(width, height, page) {
+    if (!width && !height) return page;
+
+    let viewport = page.viewport();
+    width && (viewport.width = width);
+    height && (viewport.height = height);
+
+    return page.setViewport(viewport).then(() => page);
+}
+
+function handleConsole(msg) {
+    const args = msg.args() || [];
+
+    Promise.all(args.map(a => a.jsonValue().catch(() => '')))
+        .then(args => {
+            // process stdout stub
+            let isStdout = args[0] === 'stdout:';
+            isStdout && (args = args.slice(1));
+
+            let msg = util.format(...args);
+            !isStdout && msg && (msg += '\n');
+            process.stdout.write(msg);
+        });
+}
+
+function prepareUrl(filePath) {
+    if (/^[a-zA-Z]+:\/\//.test(filePath)) {
+        // path is URL
+        return filePath;
+    }
+
+    // local path
+    let resolvedPath = path.resolve(filePath);
+    return `file://${resolvedPath}`;
+}
+
+exports.runner = function ({ file, reporter, timeout, width, height, args, executablePath, visible, polling }) {
+    return new Promise(resolve => {
+
+        // validate options
+        if (!file) {
+            throw new Error('Test page path is required.');
+        }
+
+        args = [].concat(args || []).map(arg => '--' + arg);
+        !timeout && (timeout = 60000);
```

Comment:
Nit: timeout value of 60000 is only used in one place, but I think making a const variable for timeout would have value.

Reply:
DONE